frontend/packages/console-dynamic-plugin-sdk/src/types.ts

```diff
@@ -38,7 +38,7 @@ export type CodeRef<T = any> = () => Promise<T>;
  * Infer resolved `CodeRef` properties from object `O`.
  */
 export type ResolvedCodeRefProperties<O extends {}> = {
-  [K in keyof O]: O[K] extends CodeRef<infer T> ? T : never;
+  [K in keyof O]: O[K] extends CodeRef<infer T> ? T : O[K];
```

Comment:
This change should not be necessary.

The purpose of `ResolvedCodeRefProperties` type is to narrow down (filter) the object shape.

For example:

```ts
type P = {
  foo: "bar",
  qux: () => Promise.resolve("value"), // CodeRef<string>
};

type RP = ResolvedCodeRefProperties<P>; // { qux: string }
```

Reply:
See also type `ResolvedExtension` which has two purposes:
- adapt `Extension` (base interface) to `LoadedExtension` (runtime interface)
- update extension's `properties` by replacing `CodeRef<T>` values with `T` values

Reply:
![image](https://user-images.githubusercontent.com/24938324/108254255-7430fa80-7163-11eb-8c5c-d657f198964c.png)

why would a developer prefer `scope` being `never` here?